src/handlers/touch_handler.rs

```diff
@@ -0,0 +1,218 @@
+use bevy::input::touch::{TouchInput, TouchPhase};
+use bevy::log;
+use bevy::math::Vec2;
+use bevy::window::{Window, WindowEvent};
+use smithay_client_toolkit::reexports::client::Proxy;
+use smithay_client_toolkit::{
+    delegate_touch,
+    reexports::client::{
+        Connection, QueueHandle,
+        protocol::{wl_surface::WlSurface, wl_touch::WlTouch},
+    },
+    seat::touch::TouchHandler,
+};
+
+use crate::AppSendEvent;
+use crate::smithay_windows::SmithayWindows;
+use crate::state::SmithayRunnerState; // Needed for tracking active touches
+
+impl TouchHandler for SmithayRunnerState {
+    fn down(
+        &mut self,
+        _conn: &Connection,
+        _qh: &QueueHandle<Self>,
+        _touch: &WlTouch,
+        _serial: u32,
+        _time: u32,
+        surface: WlSurface,
+        id: i32,
+        position: (f64, f64),
+    ) {
+        let entity = {
+            let smithay_windows = self.world().non_send_resource::<SmithayWindows>();
+            let window_id = surface.id();
+
+            // Find the Bevy entity associated with the Smithay surface
+            if let Some(entity) = smithay_windows.smithay_to_entity.get(&window_id).copied() {
+                entity
+            } else {
+                log::warn!("Touch down event on unknown surface: {:?}", window_id);
```

Comment:
Have a convention for log messages, up till this point all your log messages above started with lowercase letter, we should make them proper sentences starting with capitalized letter.